apps/extension/src/background/keyring/messages.ts

```diff
@@ -141,23 +146,27 @@ export class RenameAccountMsg extends Message<DerivedAccount> {
   }
 }
 
-export class SaveMnemonicMsg extends Message<AccountStore | false> {
+export class SaveAccountSecretMsg extends Message<AccountStore | false> {
   public static type(): MessageType {
-    return MessageType.SaveMnemonic;
+    return MessageType.SaveAccountSecret;
   }
 
-  constructor(public readonly words: string[], public readonly alias: string) {
+  constructor(
+    public readonly accountSecret: AccountSecret,
+    public readonly alias: string
+  ) {
     super();
   }
 
   validate(): void {
-    if (!this.words) {
-      throw new Error("A wordlist is required to save a mnemonic!");
+    if (!this.accountSecret) {
+      throw new Error("A wordlist or private key is required!");
     }
 
     if (
```

Comment:
Do you think we should validate here for the privateKey as well?

Reply:
We could at least check that it's 64 bytes, which is what our private keys are, though I'm not sure if it could be less than that...